src/core/image.js

```diff
@@ -343,6 +347,80 @@ class PDFImage {
     return { data, width, height, interpolate };
   }
 
+  static createMask({
+    imgArray,
+    width,
+    height,
+    imageIsFromDecodeStream,
+    inverseDecode,
+    interpolate,
+  }) {
+    if (
+      typeof PDFJSDev === "undefined" ||
+      PDFJSDev.test("!PRODUCTION || TESTING")
+    ) {
+      assert(
+        imgArray instanceof Uint8ClampedArray,
+        'PDFImage.createMask: Unsupported "imgArray" type.'
+      );
+    }
+
+    const isSingleOpaquePixel =
+      width === 1 &&
+      height === 1 &&
+      inverseDecode === (imgArray.length === 0 || !!(imgArray[0] & 128));
+
+    if (isSingleOpaquePixel) {
+      return { isSingleOpaquePixel };
+    }
+
+    let bitmap, bitmapPromise, data;
+    if (hasOffscreenCanvas) {
+      const canvas = new OffscreenCanvas(width, height);
+      const ctx = canvas.getContext("2d");
+      const imgData = ctx.createImageData(width, height);
+      getMaskImageData({
+        src: imgArray,
+        dest: imgData.data,
+        width,
+        height,
+        inverseDecode,
+      });
+
+      ctx.putImageData(imgData, 0, 0);
+      bitmap = canvas.transferToImageBitmap();
+    } else if (hasCreateImageBitmap) {
+      const imgData = new ImageData(width, height);
+      getMaskImageData({
+        src: imgArray,
+        dest: imgData.data,
+        width,
+        height,
+        inverseDecode,
+      });
+      bitmapPromise = createImageBitmap(imgData);
```

Comment:
I made it async but in the general case it's sync and in a very particular case it's async: I thought it was useless add any potential overhead just for a corner case.
Since now we're falling back on the old way to do I wonder if it's worth having the createBitmapImage case, wdyt ?

Reply:
> Since now we're falling back on the old way to do I wonder if it's worth having the createBitmapImage case, wdyt ?

That's an excellent point, let's just remove the `createBitmapImage`-case and keep this as a synchronous method then!